charms/kserve-controller/src/charm.py

```diff
@@ -185,10 +244,19 @@ def _on_install(self, event):
         try:
             self.unit.status = MaintenanceStatus("Creating k8s resources")
             self.k8s_resource_handler.apply()
-        except ApiError as e:
-            log.error(e)
+            self.cm_resource_handler.apply()
+        except ErrorWithStatus as err:
+            self.model.unit.status = err.status
+            log.error(f"Failed to handle {event} with error: {err}")
+            return
+        except ApiError as api_err:
+            log.error(api_err)
```

Comment:
At this point we will stay in MaintenanceStatus, correct?
Should we raise error here as well?